Добавлена таблица документов.

evgeniiruchko opened the pull request:
Создана сущность документ и ДТО.
Созданы классы и методы для получения списка ДТО документа.
Методы для получения не активных бонусов

evgeniiruchko committed:
Добавлена таблица документов.
Создана сущность документ и ДТО.
Созданы классы и методы для получения списка ДТО документа.
Методы для получения не активных бонусов

evgeniiruchko committed:
тесты к сервису документов и новым методам сервиса движений

AdrenalinV commented on src/main/java/ru/gb/lk_loyality/controllers/CounterController.java:
```diff
@@ -0,0 +1,56 @@
+        return counterService.getListCountersByPeriod(cardId, from, to);
+    }
+
+    @ExceptionHandler(InvalidDateException.class)
```
в контроллерах отдельный класс под обработку ексепшенов отведен DefaultAdvice это туда выкини

AdrenalinV commented on src/main/java/ru/gb/lk_loyality/entities/Document.java:
```diff
@@ -0,0 +1,33 @@
+    @Column(name = "card_id")
+    private Long cardId;
+
+    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```
почему OneToMany? одно начисление/расход по одному документу, или у нас  5 документов и одно начисление?

evgeniiruchko committed:
в сущности Document убрана связь
    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name="document_id", referencedColumnName = "id")
    private List<Counter> counters;

Обработка исключения при вводе некорректной даты выведена в класс исключений
